lib/pebble.js

```diff
@@ -29,136 +25,165 @@ function directionToTrend (direction) {
-      transformed.filtered = sgv.filtered;
-      transformed.unfiltered = sgv.unfiltered;
-      transformed.noise = sgv.noise;
-    }
-
-    return transformed;
-  });
-
-}
-
-function addExtraData (first, req, sbx) {
-  //for compatibility we're keeping battery and iob on the first bg, but they would be better somewhere else
-
-  var data = sbx.data;
-
-  function addDelta() {
-    var prev = data.sgvs.length >= 2 ? data.sgvs[data.sgvs.length - 2] : null;
-    var current = sbx.lastSGVEntry();
-
-    //for legacy reasons we need to return a 0 for delta if it can't be calculated
-    var deltaResult = delta.calc(prev, current, sbx);
-    first.bgdelta = deltaResult && deltaResult.scaled || 0;
-    if (req.mmol) {
-      first.bgdelta = first.bgdelta.toFixed(1);
-    }
-  }
-
-  function addBattery() {
-    if (data.devicestatus && data.devicestatus.uploaderBattery && data.devicestatus.uploaderBattery >= 0) {
-      first.battery = data.devicestatus.uploaderBattery.toString();
-    }
+function pebble (req, res) {
+  var ONE_DAY = 24 * 60 * 60 * 1000
+      , uploaderBattery
+      , treatmentResults
+      , profileResult
+      , sgvData = [ ]
+      , calData = [ ];
+
+  function scaleBg(bg) {
+      if (req.mmol) {
+          return units.mgdlToMMOL(bg);
+      } else {
+          return bg;
+      }
   }
 
-  function addIOB() {
-    if (req.iob) {
-      var iobResult = iob.calcTotal(data.treatments, data.profile, Date.now());
-      if (iobResult) {
-        first.iob = iobResult.display;
-      }
-      
-      sbx.properties.iob = iobResult;
-      var bwpResult = bwp.calc(sbx);
+  function sendData () {
+    var now = Date.now();
 
-      if (bwpResult) {
-        first.bwp = bwpResult.bolusEstimateDisplay;
-        first.bwpo = bwpResult.outcomeDisplay;
-      }
-      
+    //for compatibility we're keeping battery and iob here, but they would be better somewhere else
+    if (sgvData.length > 0) {
+        sgvData[0].battery = uploaderBattery ? "" + uploaderBattery : undefined;
+        if (req.iob) {
+            sgvData[0].iob = iob.calcTotal(treatmentResults.slice(0, 20), profileResult, new Date(now)).display;
+        }
     }
-  }
 
-  addDelta();
-  addBattery();
-  addIOB();
-}
-
-function prepareBGs (req, sbx) {
-  if (sbx.data.sgvs.length === 0) {
-    return [];
+    var result = { status: [ {now: now} ], bgs: sgvData.slice(0, req.count), cals: calData };
+    res.setHeader('content-type', 'application/json');
+    res.write(JSON.stringify(result));
+    res.end( );
   }
 
-  var bgs = mapSGVs(req, sbx);
-  addExtraData(bgs[0], req, sbx);
+    var earliest_data = Date.now() - ONE_DAY;
+
+    async.parallel({
+        devicestatus: function (callback) {
+            req.devicestatus.last(function (err, value) {
+                if (!err && value) {
+                    uploaderBattery = value.uploaderBattery;
+                } else {
+                    console.error("req.devicestatus.tail", err);
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1052564555/issues/source?bid=2452374&fileBranchId=2883753#l69)